src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -213,12 +224,13 @@ const AddCriteriaSelector: React.FC<AddCriteriaSelectorProps> = (props) => {
     options: [
       {
         label: 'Domains',
-        options: _.map((domainOption) => {
-          return {
-            value: domainOption,
+        options: _.map(
+          (domainOption) => ({
+            value: addKindToDomainOption(domainOption),
             label: domainOption.category,
-          };
-        }, datasetDetails.domainOptions),
+          }),
+          datasetDetails?.snapshotBuilderSettings?.domainOptions
```

Comment:
If `datasetDetails.snapshotBuilderSettings` isn't present, then this page is in error and shouldn't be displayed at all. Not sure what the best way to do this would be, but type-wise I think we should require that `snapshotBuilderSettings` exists at the top of the call chain and then not allow `undefined` inside the page.